src/main/java/org/betonquest/betonquest/objectives/BlockObjective.java

```diff
@@ -28,42 +30,64 @@ public class BlockObjective extends CountingObjective implements Listener {
 
     private final boolean noSafety;
 
+    private final CompoundLocation location;
+
+    private final boolean hasLocation;
+
+    private final boolean ignorecancel;
+
     public BlockObjective(final Instruction instruction) throws InstructionParseException {
         super(instruction);
         selector = instruction.getBlockSelector();
         exactMatch = instruction.hasArgument("exactMatch");
         targetAmount = instruction.getVarNum();
         noSafety = instruction.hasArgument("noSafety");
+        final String stringlocation = instruction.getOptional("loc", "false");
+        location = "false".equalsIgnoreCase(stringlocation) ? null : new CompoundLocation(instruction.getPackage(), stringlocation);
```

Comment:
you can simplify that to `instruction.getLocation(instruction.getOptional("loc"))` because most of the instruction.get[Object] accept null values.